packages/web3/test/integration/external-providers/helper.ts

```diff
@@ -0,0 +1,49 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { SupportedProviders } from 'web3-types';
+import Web3 from '../../../src/index';
+
+/**
+ * Performs basic RPC calls (like `eth_accounts`, `eth_blockNumber` and `eth_sendTransaction`)
+ * @param provider - an instance of a compatible provider
+ */
+export async function performBasicRpcCalls(provider: SupportedProviders) {
+	const web3 = new Web3(provider);
+
+	const accounts = await web3.eth.getAccounts();
+	expect(accounts).toBeDefined();
+	expect(accounts.length).toBeGreaterThan(0);
+
+	// get the last block number
+	const blockNumber0 = await web3.eth.getBlockNumber();
+	expect(typeof blockNumber0).toBe('bigint');
+
+	// send a transaction
+	const tx = web3.eth.sendTransaction({
```

Comment:
This tx is signed using node wallet accounts.
Can we add another (manual or automated ) test of injecting MM wallet and signing tx with it for:

1. verifying signing is working with injected wallet controlled accounts
2. verifying signing precedence is correct ( first local web3 wallets accounts are used/queried, after that wallet / node accounts ) for injected providers as well.

Reply:
Yes, this is a good point. And actually, there are varieties in the tests as follow:
- For Ganache provider, the mnemonic is used:
 https://github.com/web3/web3.js/pull/5652/files#diff-56116a720b3243384a647da52d1409a56641aeb8a0b6e2e31d833589de814f8eR27 

- With Truffle HDWallet, the private key is used:
 https://github.com/web3/web3.js/pull/5652/files#diff-d87ba5b36cccb886840cd3cd83128e56cc39018c6628abe9c75c4eba943ebd9fR37

- For HardHat provider, the node wallet account is used:
 https://github.com/web3/web3.js/pull/5652/files#diff-daebb1664f443048cee34bb9970cc0adcec394cecb0164d04226f09e3519db08R26

- With Infura as well as IN3 as provider, no authentication is made (because setting up in3 would complicate the test for the in3 setup and it will slow the test as well)
 https://github.com/web3/web3.js/pull/5652/files#diff-e19b31481dc158c2bff69fd1fc2cac063a9eeb82bef6f98fb4d69626ffaa6db8R29